remotes/docker/authorizer.go

```diff
@@ -304,17 +304,22 @@ func (ah *authHandler) doBearerAuth(ctx context.Context) (token, refreshToken st
 				// As of September 2017, GCR is known to return 404.
 				// As of February 2018, JFrog Artifactory is known to return 401.
 				// As of January 2022, ACR is known to return 400.
-				if (errStatus.StatusCode == 405 && to.Username != "") || errStatus.StatusCode == 404 || errStatus.StatusCode == 401 || errStatus.StatusCode == 400 {
+				switch status := errStatus.StatusCode; {
```

Comment:
Ah, looks like the example that @dcantah wrote was missing some bits;

```suggestion
				switch status := errStatus.StatusCode; status {
```

Reply:
Yea it was thrown together super quick to get across that switch looked nice here (definitely didn't test compilation 😭)